### DISCUSSION TITLE
如何解禁被拒绝弹窗提示的应用？

### DISCUSSION THREAD

**Discussion by yanhuacuo** [linuxdeepin/developer-center#4659] [General | 常规讨论]
场景是这样的，我在做一个可视化操控 Fcitx5-rime 的工具。
不可避免地，测试 RIME 的各项功能，每次 fcitx5-rime 响应时，都会在屏幕顶部显示「弹窗消息」。
由于十几个弹窗消息叠在一起，我手误点了「不再提示」后，它确实不再提示了。
但是我想恢复弹窗时，却找不到入口。
尽管「通知」中有「fcitx5」的项，但是即使「授权」打开，也不济于事。
我想，也许 DeepinV23 使用了进程名作为弹窗身份识别，但或许很多程序，并不在系统内置的「通知面板」那个 list 中。
被迫无奈，我重做了 DeepinV23 系统来完成后续开发。
希望 DeepinV23 改进这里，把所有禁掉通知的身份识别 ID 完整地在面板呈现出来。

**Comment by BLumia** [comment]
控制中心 的 通知 下应该可以控制是否允许某个应用展示通知的。如果没有对应的选项的话，可能是bug。

有稳定复现的步骤吗？

**Reply by yanhuacuo** [reply]
100%复现

sudo apt install fcitx5-rime

尝试「重新部署」时，禁掉一个弹窗。

后面，就再也找不到解禁入口了。

**Reply by BLumia** [reply]
cc @Decodetalkers

**Reply by Decodetalkers** [reply]
https://github.com/linuxdeepin/dde-session-ui/commit/427cd66821135a72140d3fdabf311556f406e8d2 之前修一个问题，没有id发送消息导致osd崩溃，给没有id的消息一个假的id,现在估计这个假的id被禁用了。。

更新，好像没有关系。。没有id发送时候应该是空的。。不会被设置上啊

**Reply by yanhuacuo** [reply]
> [linuxdeepin/dde-session-ui@427cd66](https://github.com/linuxdeepin/dde-session-ui/commit/427cd66821135a72140d3fdabf311556f406e8d2) 之前修一个问题，没有id发送消息导致osd崩溃，给没有id的消息一个假的id,现在估计这个假的id被禁用了。。
> 
> 更新，好像没有关系。。没有id发送时候应该是空的。。不会被设置上啊

请求尽快修复，目前心理阴影有半个足球场那么大。

**Reply by Decodetalkers** [reply]
rime发了个action到notify,action的意思是点击后返回给程序，程序自己处理，所以是rime自己隐藏了消息，和我们无关

**Reply by Decodetalkers** [reply]
![image](https://github.com/linuxdeepin/developer-center/assets/60290287/a2a295b4-0f05-4a11-bf19-9532bcecdbcb)

fcitx5自己有处理隐藏消息的功能，看这个文件，把这个文件删掉就可以了，不过问题是没有找到哪里取消这个的配置，你手动删掉就可以了，你可能要问下开发者把选项藏哪里了

**Reply by Decodetalkers** [reply]
找到了，fcitx5设置，addons,里面有个hidden notifications的设置，你那里面把隐藏的notification移除就好了

**Reply by yanhuacuo** [reply]
这个是 fcitx5 控件管理的，只是缺少相应的可视化工具。deepin 可以解决一下，每次「允许」fcitx5 通 知 时，把 fcitx5 这个文件清理掉。 fcitx5-rime 是纯粹的输入法模块，没有自管理通知的部分。
我在 deepin 下的开发已完成，希望 deepin 越来越好。

**Reply by Decodetalkers** [reply]
> 这个是 fcitx5 控件管理的，只是缺少相应的可视化工具。deepin 可以解决一下，每次「允许」fcitx5 通 知 时，把 fcitx5 这个文件清理掉。 fcitx5-rime 是纯粹的输入法模块，没有自管理通知的部分。 我在 deepin 下的开发已完成，希望 deepin 越来越好。


程序自定义行为，我们怎么管理？管理这个在fcitx5的控制面板里面，这个是fcitx5的notification管理，它管所有输入法的消息，我们只是把action触发返回通知fcitx5去禁用，你需要了解下notification协议，看里面action部分，action是程序自己管理，是否发消息也是它的行为

**Reply by Decodetalkers** [reply]
> 这个是 fcitx5 控件管理的，只是缺少相应的可视化工具。deepin 可以解决一下，每次「允许」fcitx5 通 知 时，把 fcitx5 这个文件清理掉。 fcitx5-rime 是纯粹的输入法模块，没有自管理通知的部分。 我在 deepin 下的开发已完成，希望 deepin 越来越好。

![image](https://github.com/linuxdeepin/developer-center/assets/60290287/4e1b357f-7466-449e-aef4-819d9109aedd)

看到这个notification了吗？fcitx5管理消息是在这里

**Reply by BLumia** [reply]
我补充下描述：

控制中心 的 通知 下可以控制是否允许某个应用展示通知，但是你这里 fcitx 是否不再弹出通知的那个选项并非 控制中心 所提供的通知管理功能，而是 fcitx 自己提供的设置项。后者并非 DDE 所能干预的选项。
